app/views/handlers/primary_person_list_collector.py

```diff
@@ -26,18 +26,23 @@ def get_context(self):
         return build_question_context(self.rendered_block, self.form)
 
     def handle_post(self):
+        list_name = self.rendered_block['for_list']
+        self.evaluate_and_update_section_status_on_list_change(list_name)
+
         if (
             self.form.data[self.rendered_block['add_or_edit_answer']['id']]
             == self.rendered_block['add_or_edit_answer']['value']
         ):
             self._is_adding = True
             self.questionnaire_store_updater.update_answers(self.form)
             self._primary_person_id = self.questionnaire_store_updater.add_primary_person(
-                self.rendered_block['for_list']
+                list_name
             )
+            self.evaluate_and_update_section_status_on_list_change(list_name)
+
             self.questionnaire_store_updater.save()
+
         else:
-            self.questionnaire_store_updater.remove_primary_person(
-                self.rendered_block['for_list']
-            )
+            self.questionnaire_store_updater.remove_primary_person(list_name)
+            self.evaluate_and_update_section_status_on_list_change(list_name)
```

Comment:
This is done before `update_answers` is called by `super().handle_post()` below.

Reply:
This is still the case @JamesGardiner ?

Reply:
Yes still the case - which means that a call to the method may incorrectly evaluate the current section. The current section status is evaluated correctly in the call to `super.handle_post()`. Spoke with @ajmaddaford about this last week and decided it's the least unfavourable option without a significant of refactoring to move `evaluate_and_update_section_status_on_list_change` to questionnaire_store_updater. The limitation on that being that a router would also need to be passed at the moment, which we want to avoid.